rnote-compose/src/shapes/arrow.rs

```diff
@@ -0,0 +1,212 @@
+    pub fn split(&self, n_splits: i32) -> Vec<Line> {
+        (0..n_splits)
+            .map(|i| {
+                let sub_start = self
+                    .start
+                    .lerp(&self.tip, f64::from(i) / f64::from(n_splits));
+                let sub_end = self
+                    .start
+                    .lerp(&self.tip, f64::from(i + 1) / f64::from(n_splits));
+
+                Line {
+                    start: sub_start,
+                    end: sub_end,
+                }
+            })
+            .collect::<Vec<Line>>()
+    }
+}
+
+/// Functions which you can use to interact with the struct.
+impl Arrow {
+    /// The min-length for `rline` and `lline`.
+    const TIP_LINES_MIN_LENGTH: f64 = 32.0;
+
+    /// Creating a new arrow with the given start and tip vectors.
+    pub fn new(start: na::Vector2<f64>, tip: na::Vector2<f64>) -> Self {
+        Self { start, tip }
+    }
+
+    /// convert the arrow to kurbo-elements.
+    pub fn to_kurbo(&self, stroke_width: Option<f64>) -> ArrowKurbo {
+        let main = kurbo::Line::new(self.start.to_kurbo_point(), self.tip.to_kurbo_point());
+        let tip_triangle = {
+            let tip = self.tip.to_kurbo_point();
+            let lline = self.compute_lline(stroke_width).to_kurbo_point();
+            let rline = self.compute_rline(stroke_width).to_kurbo_point();
+
+            kurbo::BezPath::from_vec(vec![
+                PathEl::MoveTo(lline),
+                PathEl::LineTo(tip),
+                PathEl::LineTo(rline),
+            ])
+        };
+
+        ArrowKurbo {
+            stem: main,
+            tip_triangle,
+        }
+    }
+}
+
+/// This implementation holds the functions to get the vectors `rline` and
+/// `lline`.
+impl Arrow {
+    /// The angle for `rline` and `lline` to the stem of the arrow.
+    const ANGLE: Radian = (13.0 / 16.0) * std::f64::consts::PI;
+
+    /// The default direction vector (the stem) if you can't compute one.
+    const DEFAULT_DIRECTION_VECTOR: na::Vector2<f64> = na::Vector2::new(1.0, 0.0);
+
+    /// Computes and returns `lline`.
+    /// Optionally add the stroke width to adjust the length of the line.
+    pub fn compute_lline(&self, stroke_width: Option<f64>) -> na::Vector2<f64> {
+        let vec_a = self.get_direction_vector() * Self::get_line_length(stroke_width);
+        let rotation_matrix = self.get_rotation_matrix();
+
+        rotation_matrix * vec_a + self.tip
+    }
+
+    /// Computes and returns `rline`.
+    /// Optionally add the stroke width to adjust the length of the line.
+    pub fn compute_rline(&self, stroke_width: Option<f64>) -> na::Vector2<f64> {
+        let vec_b = self.get_direction_vector() * Self::get_line_length(stroke_width);
+        let rotation_matrix = self.get_rotation_matrix().transpose();
+
+        rotation_matrix * vec_b + self.tip
+    }
+
+    /// Returns the normalized direction vector from `start` to `tip`.
+    fn get_direction_vector(&self) -> na::Vector2<f64> {
```

Comment:
I don't like the naming: It is not a getter, and should not be named as such. I am also missing "stem" in the name. More fitting would be: `compute_stem_direction_vector()`